manifest.go

```diff
@@ -60,13 +61,29 @@ func (f ManifestFile) parse() (manifest, error) {
 
 func (m manifest) path(s string) (string, error) {
 	r, ok := m[s]
-	if !ok {
-		return "", os.ErrNotExist
-	}
-	if r == "" {
+	if ok && r == "" {
 		return "", ErrEmpty
 	}
-	return r, nil
+	if ok {
+		return r, nil
+	}
+	// If path references a runfile that lies under a directory that itself is a
+	// runfile, then only the directory is listed in the manifest. Look up all
+	// prefixes of path in the manifest.
+	prefix := s
+	for {
```

Comment:
I guess you could move the test into the `for` loop:

```go
for prefix := s; prefix != ""; prefix, _ = path.Split(prefix) {
```

(untested)

Reply:
I had to choose between repeating `path.Split` in the initializer of the loop to prevent a redundant map lookup or merge the beginning of the function into the loop body, which has the side effect of doing the `ErrEmpty` check also for prefixes. The latter seemed fine to me, but please let me know what you think.

Reply:
I guess either is fine, but I think the redundant map lookup (the previous approach) is slightly cleaner (I was a bit confused by reading the new code), and the map lookup overhead shouldn't matter compared to e.g. reading the runfile.

Reply:
Restored the beginning of the function to what it was but kept your suggestion regarding the for loop initializer.